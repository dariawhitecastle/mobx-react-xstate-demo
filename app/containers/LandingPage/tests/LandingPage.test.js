/**
*
* Tests for LandingPage
*
* @see
https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
*
*/

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'
import { MemoryRouter } from 'react-router-dom'

import { LandingPage } from '..'
import { DEFAULT_LOCALE } from '../../../i18n'

const renderComponent = (props = {}) =>
  render(
    <IntlProvider locale={DEFAULT_LOCALE}>
      <MemoryRouter initialEntries={['/']}>
        <LandingPage {...props} />
      </MemoryRouter>
    </IntlProvider>,
  )

test('it renders and matches snapshot', () => {
  const { container } = renderComponent()
  expect(container).toMatchSnapshot()
})

test('It does not log to the console', () => {
  const spy = jest.spyOn(global.console, 'log')
  renderComponent()
  expect(spy).not.toHaveBeenCalled()
})
