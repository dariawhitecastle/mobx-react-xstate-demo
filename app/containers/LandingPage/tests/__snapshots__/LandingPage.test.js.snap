// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches snapshot 1`] = `
<div>
  <section
    class="LandingPagestyles__Section-sc-6xi0mq-0 bgXCnf"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 dNgQiZ"
    >
      <a
        class="StyledAnchor-sc-1rp7lwl-0 jgeOSx"
        href="/create-account"
      >
        Create account
      </a>
      <a
        class="StyledAnchor-sc-1rp7lwl-0 jgeOSx"
        href="/login"
      >
        Login
      </a>
    </div>
  </section>
</div>
`;
